Python/pywarpx/_libwarpx.py

```diff
@@ -392,7 +402,10 @@ def add_particles(species_name,
     lenux = np.size(ux)
     lenuy = np.size(uy)
     lenuz = np.size(uz)
-    lenattr = np.size(attr)
+    if type(attr) is np.ndarray:
```

Comment:
You could use `isinstance` here instead.
```suggestion
    if isinstance(attr, np.ndarray):
```

Reply:
This is a good point, the problem with the `is` comparison is it will break on arrays derived from `ndarray`, whereas `isinstance` correctly checks parent types too.